scrapyelasticsearch/scrapyelasticsearch.py

```diff
@@ -53,28 +54,37 @@ def from_crawler(cls, crawler):
         return ext
 
     def get_unique_key(self, unique_key):
-        if not isinstance(unique_key, str):
+        if isinstance(unique_key, list):
+            unique_key = unique_key[0]
+        elif not isinstance(unique_key, str):
             raise Exception('unique key must be str')
 
         return unique_key
 
     def index_item(self, item):
-        item_unique_key = item[self.settings.get('ELASTICSEARCH_UNIQ_KEY')]
+        item_unique_key = item[self.settings['ELASTICSEARCH_UNIQ_KEY']]
         unique_key = self.get_unique_key(item_unique_key)
 
-        local_id = hashlib.sha1(unique_key).hexdigest()
-        logging.debug("Generated unique key %s" % local_id)
+        item_id = hashlib.sha1(unique_key).hexdigest()
+        logging.debug('Generated unique key %s' % item_id)
 
-        self.es.index(index=self.settings.get('ELASTICSEARCH_INDEX'),
-                      doc_type=self.settings.get('ELASTICSEARCH_TYPE'),
-                      id=local_id,
-                      body=dict(item))
+        index_action = {
+            '_index': self.settings['ELASTICSEARCH_INDEX'],
+            '_type': self.settings['ELASTICSEARCH_TYPE'],
+            '_id': item_id,
+            '_source': dict(item)
+        }
+
+        self.items_buffer.append(index_action)
```

Comment:
Good point, I agree we should do the insertion here rather than waiting until the scraper is done